### DISCUSSION TITLE
Error Monitoring Feedback

### DISCUSSION THREAD

**Discussion by et** [highlight/highlight#5099] [Ideas]
Hey folks 👋 

This quarter we are investing a lot more time into improving our error monitoring product. We are looking for feedback from people who may (or may not) be currently using Highlight to monitor errors in their app to understand where we can be improving.

Ideas could include:

* Improving signal vs noise ratio
* Improving stacktrace formatting
* Code linking (e.g. to GitHub or VSCode)

But anything is fair game! Your feedback will help us drive the product going forward and makes bugs easier to solve.

**Comment by et** [comment]
## Improving GraphQL noise

GraphQL has no concept of a status code out of the box like a REST API would. The downside of this is that all flavors of errors are lumped together. So things like a validation error (e.g. backend returning "first name is required) would be reported as error which adds noise and makes it harder to find real errors (e.g. a 500 error).

**Comment by et** [comment]
## Ruby stacktraces are incomplete

Ruby stacktraces only include the line number. They don't even include the method that reported the error. We should be able to capture the source.

**Comment by et** [comment]
## Release/Version visibility on errors

If I attach a [version](https://www.highlight.io/docs/getting-started/client-sdk/replay-configuration/versioning-sessions-and-errors), I should be able to see the version somewhere on the error instance page. 

- Additionally, when viewing the error group, I want when the error was last seen/first seen.
- Similarly, if I Resolve an error, I should have the ability to select the version it was fixed in.

**Comment by et** [comment]
## Release/Version top level page

For a given version in Highlight.io, I want a top level page to get an overview of what has changed and what errors occurred in that version. To really get value here, I may need Highlight to connect to my github repository.

**Comment by stefanli** [comment]
This is not a comment on the error monitoring itself, but rather how to set it up. We're currently not using the error monitoring features as I've been dragging my feet getting source mapping setup. Not sure if I'm alone on this, but the friction of correctly setting up source map uploading is so high that I've been postponing it for months. :) This prevents me from having much use of the error monitoring since the stack traces won't tell me anything. Two ideas on this:

1. Thoroughly document everything that is involved in getting source mapping setup for different bundlers and a typical typescript stack. Not just the syncing part, but also how to generate them. This information is right now scattered and the webpack documentation for example is not great. If you guys provided something more akin to an end-to-end guide / tutorial on how to get setup, I think that'd be a big win.

2. Build more tooling to alleviate this problem. I realize this is a complex and large task (and maybe just a bad idea lol) but just throwing it in there as an idea. Any tools that minimizes the number of steps I need to take to get to source map generation and syncing would be helpful. Not sure what this would look like but maybe plugins for different bundlers that just takes care of everything.

**Comment by docenginee** [comment]
Check Documentation and FAQs:

Before reporting an issue, make sure to check the project's documentation, README file, or any FAQs that might address common problems.
Search Existing Issues:

Search the project's issue tracker on GitHub to see if someone else has already reported a similar issue. This can help you avoid duplicate reports.
Provide Detailed Information:

When reporting an issue, be as detailed as possible. Include information about the error, steps to reproduce it, and any relevant context such as operating system, browser, or environment.
Include Error Messages:

Include any error messages or stack traces you encounter. This information is crucial for developers to understand the nature of the problem.
Include Environment Information:

Specify details about your environment, such as the operating system version, browser version, programming language version, and any relevant dependencies.
Attach Screenshots or Code Snippets:

If applicable, include screenshots or code snippets that help illustrate the issue. This can provide additional context for developers to understand the problem.
Use a Clear and Descriptive Title:

Use a title that clearly summarizes the issue. A concise and descriptive title can make it easier for developers to identify and prioritize the problem.
Follow Project Guidelines:

Some projects may have specific guidelines for issue reporting. Check the project's CONTRIBUTING guidelines, if available, and follow any instructions provided.
Be Courteous and Patient:

Remember that open-source maintainers and contributors are volunteers. Be respectful and patient, and understand that it may take some time for them to address the issue.
Follow Up:

Stay engaged in the conversation. Respond to any follow-up questions or requests for additional information. This helps the maintainers address the issue more efficiently.
To report an issue on GitHub, you can typically navigate to the project's repository, go to the "Issues" tab, and click on the "New Issue" button.
